keanu-project/src/main/java/io/improbable/keanu/model/regression/RegressionModel.java

```diff
@@ -23,11 +24,16 @@
  */
 public class RegressionModel<OUTPUT> implements Model<DoubleTensor, OUTPUT> {
     private static final double DEFAULT_OBSERVATION_SIGMA = 1.0;
-
+    private final DoubleTensor inputTrainingData;
+    private final OUTPUT outputTrainingData;
+    private final ModelFitter<DoubleTensor, OUTPUT> fitter;
     private final LinearRegressionGraph<OUTPUT> modelGraph;
 
-    RegressionModel(LinearRegressionGraph<OUTPUT> modelGraph) {
+    RegressionModel(LinearRegressionGraph<OUTPUT> modelGraph, DoubleTensor inputTrainingData, OUTPUT outputTrainingData, ModelFitter<DoubleTensor, OUTPUT> fitter) {
```

Comment:
I would prefer it if the input/output data were passed as arguments to `fit` and `observe`, not stored as fields.

Reply:
The builder is where the user specifies what type of fitter they'd like. I personally don't like having the input/output data passed to the `fit` method, I'd rather have it in the builder as we require it for determining the shape of weight 
 / intercept vertices while we build the graphs. I've put the observe into the `build()` step